src/main/scala/uk/gov/hmrc/play/http/HttpResponse.scala

```diff
@@ -45,3 +46,13 @@ object HttpResponse {
     override def status: Int = responseStatus
   }
 }
+
+trait StreamingHttpResponse extends HttpResponse {
+  def bodyEnumerator: Enumerator[Array[Byte]] = ???
```

Comment:
Rather than having this new trait, maybe we could parameterise `HttpResponse`

```scala
trait HttpResponse[B] {
   def allHeaders: Map[String, Seq[String]] = ???
 
   def header(key: String) : Option[String] = allHeaders.get(key).flatMap { list => list.headOption }
 
   def status: Int = ???
 
   def body: B = ???
 }
```

which would allow `HttpResponse[String]`, `HttpResponse[JsValue]`, `HttpResponse[Enumerator[Array[Byte]]]` and even `HttpResponse[Html]`, `HttpResponse[MyCaseClass]`?

Would break some backwards-compatiblity in the process, but maybe that's a necessary evil?